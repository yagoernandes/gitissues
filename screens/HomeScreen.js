import React, { useState, useEffect } from 'react'
import { Ionicons } from '@expo/vector-icons'
import {
  Alert,
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  TextInput,
  FlatList
} from 'react-native'
import api from '../store/api'
import storage from '../store/asyncStorage'


export default function HomeScreen({ navigation }) {
  useEffect(() => {
    getAllRepos()
  }, [])

  const [state, setState] = useState({
    text: '',
    loading: true,
    repos: []
  })

  const addProject = async () => {
    setState(s => ({ ...s, loading: true }))
    const response = await api.addProject(state.text)
    getAllRepos()
  }

  const getAllRepos = async () => {
    try {
      const repos = await storage.get('REPOS')
      if (repos) {
        setState(s => ({ ...s, loading: false, repos, text: '' }))
      } else {
        console.log('Não há repositórios salvos')
        setState(s => ({ ...s, loading: false, repos: [], text: '' }))
      }
    } catch (error) {
      console.log('Erro ao recuperar os repositórios')
    }
  }

  return (
    <View style={styles.container}>
      <View style={styles.top}>
        <TextInput
          placeholder={'Adicionar novo repositório'}
          style={styles.addRepoText}
          value={state.text}
          onChangeText={text => setState(s => ({ ...s, text }))}
        />
        <TouchableOpacity
          style={styles.repoAdd}
          onPress={addProject}
          disabled={state.loading}>
          <Ionicons name='ios-add' size={40} color='black' />
        </TouchableOpacity>
      </View>
      <View style={styles.list}>
        {state.loading ? (
          <Text style={styles.loadingText}>Carregando</Text>
        ) : (
          <ScrollView
            style={styles.container}
            contentContainerStyle={styles.contentContainer}>
            {/* {state.repos.map(repo => (
              <Item navigate={navigation.navigate} repo={repo} key={repo.id} />
            ))} */}
            <FlatList
              data={state.repos}
              renderItem={({ item }) => (
                <Item
                  navigate={navigation.navigate}
                  repo={item}
                  key={item.id}
                />
              )}
              keyExtractor={(item, index) => String(item.id)}
              onRefresh={getAllRepos}
              refreshing={state.loading}
            />
          </ScrollView>
        )}
      </View>
    </View>
  )
}

function Item({ navigate, repo }) {
  return (
    <TouchableOpacity
      onPress={() => {
        navigate('Issues', repo)
      }}>
      <View style={styles.item}>
        <View style={styles.avatar}>
          <Image source={{ uri: repo.avatar }} style={styles.image} />
        </View>
        <View style={styles.repoText}>
          <Text style={styles.repoTitle} numberOfLines={1}>
            {repo.name}
          </Text>
          <Text style={styles.repoSubtitle} numberOfLines={1}>
            {repo.login}
          </Text>
        </View>
        <View style={styles.itemIcon}>
          <Ionicons name='ios-arrow-forward' size={32} color='#CCC' />
        </View>
      </View>
    </TouchableOpacity>
  )
}

HomeScreen.navigationOptions = {
  header: null
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#EEE'
  },
  top: {
    paddingVertical: 15,
    marginHorizontal: 20,
    display: 'flex',
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: '#CCC'
  },
  addRepoText: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    paddingVertical: 5,
    paddingHorizontal: 10,
    flex: 1
  },
  repoAdd: {
    marginLeft: 15
  },
  list: {
    flex: 1
  },
  loadingText: {
    paddingTop: 15,
    textAlign: 'center'
  },
  item: {
    display: 'flex',
    flexDirection: 'row',
    backgroundColor: '#FFF',
    borderRadius: 5,
    padding: 10,
    marginBottom: 15
  },
  image: {
    width: 60,
    height: 60,
    resizeMode: 'contain',
    borderRadius: 100,
    marginRight: 15
  },
  contentContainer: {
    paddingTop: 15,
    marginHorizontal: 20
  },
  repoText: {
    flex: 1,
    justifyContent: 'center'
  },
  repoTitle: {
    fontWeight: '700',
    fontSize: 18,
    marginBottom: 3
  },
  repoSubtitle: {
    color: '#CCC',
    fontSize: 14
  },
  itemIcon: {
    paddingRight: 10,
    justifyContent: 'center',
    marginLeft: 10
  },

  developmentModeText: {
    marginBottom: 20,
    color: 'rgba(0,0,0,0.4)',
    fontSize: 14,
    lineHeight: 19,
    textAlign: 'center'
  },
  welcomeContainer: {
    alignItems: 'center',
    marginTop: 10,
    marginBottom: 20
  },
  getStartedContainer: {
    alignItems: 'center',
    marginHorizontal: 50
  },
  homeScreenFilename: {
    marginVertical: 7
  },
  codeHighlightText: {
    color: 'rgba(96,100,109, 0.8)'
  },
  codeHighlightContainer: {
    backgroundColor: 'rgba(0,0,0,0.05)',
    borderRadius: 3,
    paddingHorizontal: 4
  },
  getStartedText: {
    fontSize: 17,
    color: 'rgba(96,100,109, 1)',
    lineHeight: 24,
    textAlign: 'center'
  },
  tabBarInfoContainer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    ...Platform.select({
      ios: {
        shadowColor: 'black',
        shadowOffset: { width: 0, height: -3 },
        shadowOpacity: 0.1,
        shadowRadius: 3
      },
      android: {
        elevation: 20
      }
    }),
    alignItems: 'center',
    backgroundColor: '#fbfbfb',
    paddingVertical: 20
  },
  tabBarInfoText: {
    fontSize: 17,
    color: 'rgba(96,100,109, 1)',
    textAlign: 'center'
  },
  navigationFilename: {
    marginTop: 5
  },
  helpContainer: {
    marginTop: 15,
    alignItems: 'center'
  },
  helpLink: {
    paddingVertical: 15
  },
  helpLinkText: {
    fontSize: 14,
    color: '#2e78b7'
  }
})
