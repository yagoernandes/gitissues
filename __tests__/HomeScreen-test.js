import React from 'react'
import NavigationTestUtils from 'react-navigation/NavigationTestUtils'
import renderer from 'react-test-renderer'
// import { render, fireEvent } from 'react-testing-library'
import HomeScreen from '../screens/HomeScreen'

jest.mock('expo', () => ({
  AppLoading: 'AppLoading'
}))

jest.mock('../navigation/AppNavigator', () => 'AppNavigator')

describe('HomeScreen', () => {
  jest.useFakeTimers()

  beforeEach(() => {
    NavigationTestUtils.resetInternalState()
  })

  it(`renders the home screen`, () => {
    const tree = renderer.create(<HomeScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })

})