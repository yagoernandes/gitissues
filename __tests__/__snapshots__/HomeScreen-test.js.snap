// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders the home screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EEE",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#CCC",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "marginHorizontal": 20,
        "paddingVertical": 15,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Adicionar novo repositÃ³rio"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "flex": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": 15,
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "paddingTop": 15,
          "textAlign": "center",
        }
      }
    >
      Carregando
    </Text>
  </View>
</View>
`;
